sdk/cosmos/azure-cosmos/test/test_auto_scale.py

```diff
@@ -0,0 +1,141 @@
+class AutoScaleTest(unittest.TestCase):
+    host = test_config._test_config.host
+    masterKey = test_config._test_config.masterKey
+    connectionPolicy = test_config._test_config.connectionPolicy
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
+                cls.host == '[YOUR_ENDPOINT_HERE]'):
+            raise Exception(
+                "You must specify your Azure Cosmos account values for "
+                "'masterKey' and 'host' at the top of this class to run the "
+                "tests.")
+
+        cls.client = cosmos_client.CosmosClient(cls.host, cls.masterKey, consistency_level="Session",
+                                                connection_policy=cls.connectionPolicy)
+        cls.created_database = cls.client.create_database(test_config._test_config.TEST_DATABASE_ID)
+
+    def test_auto_scale_max_throughput(self):
+        created_container = self.created_database.create_container(
+            id='container_with_auto_scale_settings',
+            partition_key=PartitionKey(path="/id"),
+            offer_throughput=ThroughputProperties(auto_scale_max_throughput=7000, auto_scale_increment_percent=0)
+
+        )
+        created_container_properties = created_container.get_throughput()
+        # Testing the input value of the max_throughput
+        self.assertEqual(
+            created_container_properties.auto_scale_max_throughput, 7000)
+
+        self.created_database.delete_container(created_container)
+
+        # Testing the incorrect passing of an input value of the max_throughput to verify negative behavior
+        with pytest.raises(exceptions.CosmosHttpResponseError) as e:
+            created_container = self.created_database.create_container(
+                id='container_with_wrong_auto_scale_settings',
+                partition_key=PartitionKey(path="/id"),
+                offer_throughput=ThroughputProperties(auto_scale_max_throughput=-200, auto_scale_increment_percent=0))
+        assert "Requested throughput -200 is less than required minimum throughput 1000" in str(e.value)
+
+    def test_auto_scale_increment_percent(self):
+        created_container = self.created_database.create_container(
+            id='container_with_auto_scale_settings',
+            partition_key=PartitionKey(path="/id"),
+            offer_throughput=ThroughputProperties(auto_scale_max_throughput=5000, auto_scale_increment_percent=1)
+
+        )
+        created_container_properties = created_container.get_throughput()
+        # Testing the input value of the increment_percentage
+        self.assertEqual(
+            created_container_properties.auto_scale_increment_percent, 1)
+
+        self.created_database.delete_container(created_container)
+
+    def test_auto_scale_settings(self):
+        # Testing for wrong attributes for the auto_scale_settings in the created container
+        with pytest.raises(AssertionError) as e:
```

Comment:
This should raise a `TypeError` - I didn't see in the code where the `AssertionError` would be coming from?